blockchain/chain/src/store/chain_store.rs

```diff
@@ -301,6 +301,50 @@ where
         }
     }
 
+    pub async fn validate_tipset_checkpoints(
+        &self,
+        from: Arc<Tipset>,
+        to: ChainEpoch,
+    ) -> Result<(), Error> {
+        info!("Validating tipset checkpoint hashes from: {}", from.epoch());
+        if to > from.epoch() {
+            return Err(Error::Other(
+                "Looking for tipset with height greater than start point".to_string(),
+            ));
+        }
+
+        // filter tipset checkpoints that are within the snapshot's range.
+        let mut hashes: HashSet<String> = checkpoint_tipsets::TIPSET_CHECKPOINTS
+            .keys()
+            .filter(|s| s.0 <= from.epoch())
+            .map(|s| s.1.to_string())
+            .collect();
```

Comment:
A checkpoint in the future (ie with an epoch in the future) should be an error, I think.

Reply:
What if some has a snapshot file that contains that "checkpoint in future", should we still consider that as an error?

Reply:
I'm not sure what you mean. My point is that the calibnet checkpoint for epoch 1405400 is definitely wrong since calibnet is currently at epoch 61000.

Reply:
Removed chain_epoch. Changed validation fn to only consider current "network" checkpoints